src/Pages/Search/Search.js

```diff
@@ -1,2 +1,299 @@
+    <SearchPage>
+      <div className="innerSearch">
+        <InputWrap>
+          <div className="innerInput">
+            <div className="selectArea open">
+              <div className="selectedPart">
+                <ul>
+                  <li>
+                    <a href="#" data-value="all">
+                      전체
+                    </a>
+                  </li>
+                  <li>
+                    <a href="#" data-value="title">
+                      제목
+                    </a>
+                  </li>
+                  <li>
+                    <a href="#" data-value="author">
+                      작가
+                    </a>
+                  </li>
+                  <li>
+                    <a href="#" data-value="publisher">
+                      출판사
+                    </a>
+                  </li>
+                </ul>
+              </div>
+            </div>
+            <div className="inputArea">
+              <FaSearch className="searchIcon" />
+              <input type="text" placeholder="검색어를 입력하세요." />
+            </div>
+          </div>
+        </InputWrap>
+        <RecommendWrap>
+          <h2>밀리 추천 검색어</h2>
+          <div className="searchRecommend">
+            <ul>
+              <li>
+                <FaSearch className="searchIcon" />
+                <a href={`/search/result/최준?type=all&category=1`}>
+                  지금 주목 받는
+                  <strong> 최 준 작가</strong>
+                </a>
+              </li>
+              <li>
+                <FaSearch className="searchIcon" />
+                <a href={`/search/result/개발철학?type=all&category=1`}>
+                  세상을 움직이는
+                  <strong> 개발자의 철학</strong>
+                </a>
+              </li>
+              <li>
+                <FaSearch className="searchIcon" />
+                <a href={`/search/result/아무튼?type=all&category=1`}>
+                  좋아하는 게 너무 많아,
+                  <strong> 아무튼 시리즈</strong>
+                </a>
+              </li>
+              <li>
+                <FaSearch className="searchIcon" />
+                <a href={`/search/result/프로그래밍?type=all&category=1`}>
+                  누구나 쉽게 하는
+                  <strong> 프로그래밍</strong>
+                </a>
+              </li>
+            </ul>
+          </div>
+        </RecommendWrap>
+        <Banner>
+          <a href="https://www.millie.co.kr/viewfinder/millie_best.html">
+            <h2>지금 인기 있는 책 TOP 30</h2>
+            <p>밀리 회원이 많이 읽은 책을 살펴보세요.</p>
+          </a>
+        </Banner>
+        <Books>
+          <h2>밀리 도서 카테고리</h2>
+          <ul className="bottom">
+            {categories.map((category) => {
+              return (
+                <li id={category?.id}>
+                  <img
+                    className="bookCover"
+                    src={category?.image}
+                    alt="Category"
+                  />
+                  <span>{category?.name}</span>
+                </li>
```

Comment:
- id selector를 사용한 이유가 궁금합니다
- 제 생각에는 클래스 셀렉터 대신 img 태그 셀렉터를 사용하면 불필요한 클래스명 부여를 피하고, alt 속성을 이용해 해당 이미지에 대한설명을 부여하는 것이 더 좋아보입니다.
- 구조분해 할당을 이용해볼 수 있습니다
- optional chaining이 개별 요소에 부여되기 보다는 상위 요소에 하나만 주어지는게 자연스럽습니다
```suggestion
            {categories?.map(({ id, image, name }) => {
              return (
                <li>
                  <img alt="book cover" src={image} />
                  <span>{name}</span>
                </li>
```

Reply:
* key={category?.id} 라고 써야하는 것을 id selector로 잘못 사용했어요. 해당 부분은 destructuring 후에 `key={id}`으로 변경했습니다!
* ul과 img 태그 모두 클래스명 부여가 불필요한 것 같아서 삭제하고, img alt 속성으로 설명을 보충했습니다
* 구조 분해 할당하고, optional chaining은 map 앞에 한 번만 사용하도록 변경했습니다!

변경한 코드 >>>
~~~js
<ul>
            {categories?.map(({ id, image, name }) => {
              return (
                <li key={id} onClick={() => goToSubCatePage(id)}>
                  <img src={image} alt="Book cover for category" />
                  <span>{name}</span>
                </li>
              );
            })}
          </ul>
~~~